perfmetrics/scripts/testing_on_gke/examples/utils/gsheet.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tempfile
+from typing import Tuple
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+from utils.utils import run_command
+
+_SCOPES = ['https://www.googleapis.com/auth/spreadsheets']
+
+
+def _get_sheets_service_client(localServiceAccountKeyFile):
+  creds = service_account.Credentials.from_service_account_file(
+      localServiceAccountKeyFile, scopes=_SCOPES
+  )
+  # Alternatively, use from_service_account_info for client-creation,
+  # documented at
+  # https://google-auth.readthedocs.io/en/master/reference/google.oauth2.service_account.html
+  # .
+  client = build('sheets', 'v4', credentials=creds)
+  return client
+
+
+def download_gcs_object_locally(gcsObjectUri: str) -> str:
+  """Downloads the given gcs file-object to a temporary local file (collision-free) and returns the full-path of this local-file.
+
+  gcsObjectUri is of the form: gs://<bucket-name>/object-name .
+  On failure, returned int will be non-zero.
+
+  Caller has to delete the tempfile after usage if a proper tempfile was
+  returned, to avoid disk-leak.
+  """
+  if not gcsObjectUri.startswith('gs:'):
+    raise ValueError(
+        f'Passed input {gcsObjectUri} is not proper. Expected:'
+        ' gs://<bucket>/<object-name>'
+    )
+  with tempfile.NamedTemporaryFile(mode='w+b', dir='/tmp', delete=False) as fp:
+    returncode = run_command(f'gcloud storage cp {gcsObjectUri} {fp.name}')
+    if returncode == 0:
+      return fp.name
+    else:
+      raise Exception(
+          f'failed to copy gcs object {gcsObjectUri} to local-file {fp.name}:'
+          f' returncode={returncode}. Deleting tempfile {fp.name}...'
+      )
+      os.remove(fp.name)
+
+
+def append_data_to_gsheet(
```

Comment:
Why didn't we use this - https://github.com/GoogleCloudPlatform/gcsfuse/blob/44802212cfdb0b8e7353d89dc7645bb92dfbddf3/perfmetrics/scripts/gsheet/gsheet.py#L31?